src/eddymotion/model/dipy.py

```diff
@@ -351,9 +332,58 @@ def compute_spherical_covariance(
     return np.where(theta <= a, 1 - 3 * (theta / a) ** 2 + 2 * (theta / a) ** 3, 0)
 
 
+def compute_derivative(
+    theta: np.ndarray,
+    kernel: np.ndarray,
+    weighting: str,
+    params: dict[float],
+):
+    """
+    Compute the analytical derivative of the kernel.
+
+    Parameters
+    ----------
+    theta : :obj:`~numpy.ndarray`
+        Pairwise angles across diffusion gradient encoding directions.
+    kernel : :obj:`~numpy.ndarray`
+        Current kernel.
+    weighting : :obj:`str`
+        The kind of kernel which derivatives will be calculated.
+    params : :obj:`dict`
+        Current parameter values.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Gradients of the kernel.
+
+    """
+
+    n_partials = len(params)
+
+    a = params.pop("a")
+    lambda_s = params.pop("lambda_s")
+
+    min_angles = theta > a
+
+    if weighting == "spherical":
+        deriv_a = (3 * theta[min_angles] / a**2) - (1.5 * (theta[min_angles] / a) ** 2) / a
+    elif weighting == "exponential":
+        deriv_a = theta[min_angles] * a * np.exp(-theta[min_angles] / a)
+    else:
```

Comment:
Also, and although this is not important now, IMO it would make sense to have these computed in separate functions, much like their integral counterparts are available in separate functions.

Reply:
Yeah, I'm struggling with this. On the one hand, we want consistency (i.e., implement the kernel and the gradient either in separate or single functions, but both the same). On the other hand, I can't foresee more kernels being added, so it looks an overshoot to have functions for each kernel.